components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2870,4 +2878,59 @@ void RewardsServiceImpl::OnDeleteActivityInfo(
   }
 }
 
+void RewardsServiceImpl::Reset(ResetStateCallback callback) {
+  notification_service_->DeleteAllNotifications();
+  bat_ledger_->DeleteStateFiles(
+      base::BindOnce(&RewardsServiceImpl::OnReset,
+        AsWeakPtr(), std::move(callback)));
+}
+
+void RewardsServiceImpl::OnReset(
+    ResetStateCallback callback,
+    bool success) {
+  std::move(callback).Run(success);
+  for (auto& observer_ : observers_) {
+    observer_.OnReset(this);
+  }
+}
+
+void RewardsServiceImpl::DeleteClientStateFiles(
+    ledger::OnResetClientStateCallback callback) {
+#if defined(OS_WIN)
+  const base::FilePath::StringType
```

Comment:
we should be clearing the database, not deleting the db files

Reply:
There are is no db file pre-opt-in

Reply:
That doesn't make any difference. First of all it's not safe or reliable to try to delete the db files and second the existence of the files is irrelevant. What matters is whether there is data in them or not.

Reply:
Would the state files (`ledger_state`, `publisher_state`, `confirmations.json`, etc) remain as well or be deleted?